pkg/templates/kubeadm/v1beta1/kubeadm.go

```diff
@@ -35,9 +35,19 @@ import (
 func NewConfig(ctx *util.Context, host kubeoneapi.HostConfig) ([]runtime.Object, error) {
 	cluster := ctx.Cluster
 
+	nodeIP := host.PrivateAddress
+	if nodeIP == "" {
+		nodeIP = host.PublicAddress
+	}
+
 	nodeRegistration := kubeadmv1beta1.NodeRegistrationOptions{
-		Name:             host.Hostname,
-		KubeletExtraArgs: map[string]string{"node-ip": host.PrivateAddress},
+		Name: host.Hostname,
+		KubeletExtraArgs: map[string]string{
+			"anonymous-auth":      "false",
+			"node-ip":             nodeIP,
+			"read-only-port":      "0",
+			"rotate-certificates": "true",
```

Comment:
How does this work?

Reply:
via CertificateRequests. https://kubernetes.io/docs/tasks/tls/certificate-rotation/ and although RotateKubeletClientCertificate feature gate is not enabled, it's a parity match with machine-controller inited worker nodes.

P.S.
probably we'd want to enable `RotateKubeletClientCertificate` at some point in the future.

Reply:
Please create a follow-up issue for this.

Reply:
feature flags actually are [enabled by default](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping/#certificate-rotation
)

Reply:
What about this note:
> Note: The CSR approving controllers implemented in core Kubernetes do not approve node serving certificates for security reasons. To use RotateKubeletServerCertificate operators need to run a custom approving controller, or manually approve the serving certificate requests.

Does it affect us?